slapo/model_dialect/deepspeed/engine.py

```diff
@@ -22,7 +22,11 @@ def init_ds_engine(model, **kwargs):
         raise ValueError("DeepSpeed config not provided.")
     mpu = kwargs.get("topology", None)
     if mpu is not None and isinstance(mpu, PipeModelDataParallelTopology):
-        mpu = PipelineParallelGrid(topology=mpu)
+        if mpu.get_dim("pipe") <= 1:
```

Comment:
it could be 0?

Reply:
Not really, just for separating the conditions for pipeline and no pipeline in a binary form.